src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -17,33 +17,43 @@ public class ShoppingCartServiceImpl implements ShoppingCartService {
     private ShoppingCartDao shoppingCartDao;
     @Inject
     private TicketDao ticketDao;
+    @Inject
+    private UserDao userDao;
 
     @Override
     public void addSession(MovieSession movieSession, User user) {
-        Ticket newTicket = new Ticket();
-        newTicket.setUser(user);
-        newTicket.setMovieSession(movieSession);
-
-        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
-        shoppingCart.getTickets().add(ticketDao.add(newTicket));
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
+        ticketDao.add(ticket);
+        ShoppingCart shoppingCart = getByUser(user);
+        shoppingCart.getTickets().add(ticket);
         shoppingCartDao.update(shoppingCart);
     }
 
     @Override
     public ShoppingCart getByUser(User user) {
-        return shoppingCartDao.getByUser(user);
+        return shoppingCartDao.getByUser(user).orElseThrow(
+                () -> new RuntimeException("ShoppingCart not found")
+        );
     }
 
     @Override
     public void registerNewShoppingCart(User user) {
+        if (userDao.get(user.getEmail()).isEmpty()) {
```

Comment:
The check for user existence should be done using a method that verifies the user's presence in a more robust way, possibly by checking the user's ID instead of email, to avoid potential issues with email uniqueness.